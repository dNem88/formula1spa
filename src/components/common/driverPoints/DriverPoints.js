import React from 'react'
import styles from './driverPoints.module.css'

function DriverPoints({driver, position, team, points}) {
    const fullName = driver.split(' ')
    const firstName = fullName[0]
    const lastName = fullName[1]
    return (
        <div className={styles['driver-row']}>
            <p>{position}</p>
            <p>{firstName}</p>
            <p>{lastName}</p>
            <p>{team}</p>
            <p>{points}</p>
        </div>
    )
}

export default DriverPoints