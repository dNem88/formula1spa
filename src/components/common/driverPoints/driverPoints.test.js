import React from 'react'
import renderer from 'react-test-renderer'
import DriverPoints from './DriverPoints'
import { MemoryRouter } from 'react-router-dom'
const mockProps = {
    driver: 'Lewis Hamilton',
    position: 2,
    team: 'Mercedes',
    points: 100
}

describe('Testing DriverPoints', () => {
    it('Render data properly', () => {
        const snapshot = renderer.create(
            <MemoryRouter>
                <DriverPoints {...mockProps}/>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('div')
        expect(snapshot.props.className).toBe('driver-row')
        expect(snapshot.children[0].type).toBe('p')
        expect(snapshot.children[0].children[0]).toBe('2')
        expect(snapshot.children[2].type).toBe('p')
        expect(snapshot.children[2].children[0]).toBe('Hamilton')
        expect(snapshot.children[4].type).toBe('p')
        expect(snapshot.children[4].children[0]).toBe('100')
        expect(snapshot.children[1].type).toBe('p')
        expect(snapshot.children[1].children[0]).toBe('Lewis')
        expect(snapshot.children[3].type).toBe('p')
        expect(snapshot.children[3].children[0]).toBe('Mercedes')
    })
})
