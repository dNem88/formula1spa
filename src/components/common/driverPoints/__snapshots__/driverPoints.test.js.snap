// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DriverPoints Render data properly 1`] = `
<div
  className="driver-row"
>
  <span>
    2
  </span>
  <p>
    Lewis
  </p>
  <span>
    Hamilton
  </span>
  <p>
    Mercedes
  </p>
  <span>
    100
  </span>
</div>
`;
